Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java

```diff
@@ -12,60 +12,62 @@
 package org.kitodo.production.services.command;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
+import org.apache.commons.lang.text.StrTokenizer;
+
 public class MetadataScript {
 
-    private String goal;
-    private String root;
+    private Map<String, String> parameters;
+    private String metadataKey;
+    private String valueSource;
     private String typeTarget;
+    private String variable;
     private List<String> values = new ArrayList<>();
 
     /**
      * Creates a MetadataScript with given command.
      * @param command the given command.
      */
     public MetadataScript(String command) {
-        String metadataCommand = command;
-        String typeCommand = "";
-        if (command.contains("+")) {
-            String[] commandParts = command.split("\\+");
-            metadataCommand = commandParts[0];
-            typeCommand = commandParts[1];
-        }
-        if (metadataCommand.contains("=")) {
-            String[] metadataParts = metadataCommand.split("=");
-            goal = metadataParts[0];
-            String rootOrValue = metadataParts[1];
-            if (rootOrValue.startsWith("@") || rootOrValue.startsWith("$")) {
-                root = rootOrValue;
-            } else {
-                values.add(rootOrValue);
+        this.parameters = new HashMap<>();
```

Comment:
`parameters` isn’t used outside of the function. Please replace the class field by a local variable.